diff -Naur live.orig/liveMedia/MPEG1or2VideoRTPSink.cpp live.step1/liveMedia/MPEG1or2VideoRTPSink.cpp
--- live.orig/liveMedia/MPEG1or2VideoRTPSink.cpp	2025-11-07 05:17:38.000000000 +0900
+++ live.step1/liveMedia/MPEG1or2VideoRTPSink.cpp	2025-11-30 15:19:48.723247124 +0900
@@ -19,6 +19,7 @@
 // Implementation
 
 #include "MPEG1or2VideoRTPSink.hh"
+#include <stdint.h>
 #include "MPEG1or2VideoStreamFramer.hh"
 
 MPEG1or2VideoRTPSink::MPEG1or2VideoRTPSink(UsageEnvironment& env, Groupsock* RTPgs)
@@ -121,7 +122,7 @@
     } else {
       // The first 4 bytes aren't a code that we recognize.
       envir() << "Warning: MPEG1or2VideoRTPSink::doSpecialFrameHandling saw strange first 4 bytes "
-	      << (void*)startCode << ", but we're not a fragment\n";
+	      << (void *)(uintptr_t)startCode << ", but we're not a fragment\n";
     }
   } else {
     // We're a fragment (other than the first) of a slice.
diff -Naur live.orig/liveMedia/MPEG1or2VideoStreamFramer.cpp live.step1/liveMedia/MPEG1or2VideoStreamFramer.cpp
--- live.orig/liveMedia/MPEG1or2VideoStreamFramer.cpp	2025-11-07 05:17:38.000000000 +0900
+++ live.step1/liveMedia/MPEG1or2VideoStreamFramer.cpp	2025-11-30 15:19:48.723247124 +0900
@@ -20,6 +20,7 @@
 // Implementation
 
 #include "MPEG1or2VideoStreamFramer.hh"
+#include <stdint.h>
 #include "MPEGVideoStreamParser.hh"
 #include <string.h>
 
@@ -460,7 +461,7 @@
     }
     default: {
       usingSource()->envir() << "MPEG1or2VideoStreamParser::parseSlice(): Saw unexpected code "
-			    << (void*)next4Bytes << "\n";
+			    << (void *)(uintptr_t)next4Bytes << "\n";
       setParseState(PARSING_SLICE); // the safest way to recover...
       break;
     }
diff -Naur live.orig/liveMedia/MPEG2TransportFileServerMediaSubsession.cpp live.step1/liveMedia/MPEG2TransportFileServerMediaSubsession.cpp
--- live.orig/liveMedia/MPEG2TransportFileServerMediaSubsession.cpp	2025-11-07 05:17:38.000000000 +0900
+++ live.step1/liveMedia/MPEG2TransportFileServerMediaSubsession.cpp	2025-11-30 15:19:48.724247165 +0900
@@ -20,6 +20,7 @@
 // Implementation
 
 #include "MPEG2TransportFileServerMediaSubsession.hh"
+#include <stdint.h>
 #include "SimpleRTPSink.hh"
 
 MPEG2TransportFileServerMediaSubsession*
@@ -182,7 +183,7 @@
     ClientTrickPlayState* client = lookupClient(clientSessionId);
     if (client == NULL) {
       client = newClientTrickPlayState();
-      fClientSessionHashTable->Add((char const*)clientSessionId, client);
+      fClientSessionHashTable->Add((char const*)(uintptr_t)clientSessionId, client);
     }
     client->setSource(framer);
   }
@@ -216,7 +217,7 @@
 
 ClientTrickPlayState* MPEG2TransportFileServerMediaSubsession
 ::lookupClient(unsigned clientSessionId) {
-  return (ClientTrickPlayState*)(fClientSessionHashTable->Lookup((char const*)clientSessionId));
+  return (ClientTrickPlayState*)(fClientSessionHashTable->Lookup((char const*)(uintptr_t)clientSessionId));
 }
 
 
diff -Naur live.orig/liveMedia/MPEG2TransportStreamFramer.cpp live.step1/liveMedia/MPEG2TransportStreamFramer.cpp
--- live.orig/liveMedia/MPEG2TransportStreamFramer.cpp	2025-11-07 05:17:38.000000000 +0900
+++ live.step1/liveMedia/MPEG2TransportStreamFramer.cpp	2025-11-30 15:19:48.724247165 +0900
@@ -21,6 +21,7 @@
 // Implementation
 
 #include "MPEG2TransportStreamFramer.hh"
+#include <stdint.h>
 #include <GroupsockHelper.hh> // for "gettimeofday()"
 
 #define TRANSPORT_PACKET_SIZE 188
@@ -230,12 +231,12 @@
   unsigned pid = ((pkt[1]&0x1F)<<8) | pkt[2];
 
   // Check whether we already have a record of a PCR for this PID:
-  PIDStatus* pidStatus = (PIDStatus*)(fPIDStatusTable->Lookup((char*)pid));
+  PIDStatus* pidStatus = (PIDStatus*)(fPIDStatusTable->Lookup((char *)(uintptr_t)pid));
 
   if (pidStatus == NULL) {
     // We're seeing this PID's PCR for the first time:
     pidStatus = new PIDStatus(clock, timeNow);
-    fPIDStatusTable->Add((char*)pid, pidStatus);
+    fPIDStatusTable->Add((char *)(uintptr_t)pid, pidStatus);
 #ifdef DEBUG_PCR
     fprintf(stderr, "PID 0x%x, FIRST PCR 0x%08x+%d:%03x == %f @ %f, pkt #%lu\n", pid, pcrBaseHigh, pkt[10]>>7, pcrExt, clock, timeNow, fTSPacketCount);
 #endif
diff -Naur live.orig/liveMedia/MPEG4VideoStreamFramer.cpp live.step1/liveMedia/MPEG4VideoStreamFramer.cpp
--- live.orig/liveMedia/MPEG4VideoStreamFramer.cpp	2025-11-07 05:17:38.000000000 +0900
+++ live.step1/liveMedia/MPEG4VideoStreamFramer.cpp	2025-11-30 15:19:48.724247165 +0900
@@ -24,6 +24,7 @@
 // Implementation
 
 #include "MPEG4VideoStreamFramer.hh"
+#include <stdint.h>
 #include "MPEGVideoStreamParser.hh"
 #include "MPEG4LATMAudioRTPSource.hh" // for "parseGeneralConfigStr()"
 #include <string.h>
@@ -652,7 +653,7 @@
       setParseState(PARSING_VIDEO_OBJECT_PLANE);
     } else {
       usingSource()->envir() << "MPEG4VideoStreamParser::parseVideoObjectPlane(): Saw unexpected code "
-			     << (void*)next4Bytes << "\n";
+			     << (void *)(uintptr_t)next4Bytes << "\n";
       setParseState(PARSING_VIDEO_OBJECT_PLANE); // the safest way to recover...
     }
     break;
diff -Naur live.orig/liveMedia/MatroskaFile.cpp live.step1/liveMedia/MatroskaFile.cpp
--- live.orig/liveMedia/MatroskaFile.cpp	2025-11-07 05:17:38.000000000 +0900
+++ live.step1/liveMedia/MatroskaFile.cpp	2025-11-30 15:19:48.724247165 +0900
@@ -19,6 +19,7 @@
 // Implementation
 
 #include "MatroskaFileParser.hh"
+#include <stdint.h>
 #include "MatroskaDemuxedTrack.hh"
 #include <ByteStreamFileSource.hh>
 #include <H264VideoStreamDiscreteFramer.hh>
@@ -151,8 +152,8 @@
   delete fDemuxesTable;
   delete fTrackTable;
 
-  delete[] (char*)fPreferredLanguage;
-  delete[] (char*)fFileName;
+  delete[] (char *)(uintptr_t)fPreferredLanguage;
+  delete[] (char *)(uintptr_t)fFileName;
 }
 
 void MatroskaFile::handleEndOfTrackHeaderParsing(void* clientData) {
@@ -249,15 +250,15 @@
   demuxRecord->onDeletionFunc = onDeletionFunc;
   demuxRecord->objectToNotify = objectToNotify;
 
-  fDemuxesTable->Add((char const*)demux, demuxRecord);
+  fDemuxesTable->Add((char const*)(uintptr_t)demux, demuxRecord);
 
   return demux;
 }
 
 void MatroskaFile::removeDemux(MatroskaDemux* demux) {
-  DemuxRecord* demuxRecord = (DemuxRecord*)(fDemuxesTable->Lookup((char const*)demux));
+  DemuxRecord* demuxRecord = (DemuxRecord*)(fDemuxesTable->Lookup((char const*)(uintptr_t)demux));
   if (demuxRecord != NULL) {
-    fDemuxesTable->Remove((char const*)demux);
+    fDemuxesTable->Remove((char const*)(uintptr_t)demux);
 
     if (demuxRecord->onDeletionFunc != NULL) {
       (*demuxRecord->onDeletionFunc)(demuxRecord->objectToNotify, demux);
@@ -703,8 +704,8 @@
 
       getH264ConfigData(track, sps, spsSize, pps, ppsSize);
 
-      char* sps_base64 = base64Encode((char*)sps, spsSize);
-      char* pps_base64 = base64Encode((char*)pps, ppsSize);
+      char* sps_base64 = base64Encode((char *)(uintptr_t)sps, spsSize);
+      char* pps_base64 = base64Encode((char *)(uintptr_t)pps, ppsSize);
       delete[] sps; delete[] pps;
 
       char* sPropParameterSetsStr
@@ -725,9 +726,9 @@
 
       getH265ConfigData(track, vps, vpsSize, sps, spsSize, pps, ppsSize);
 
-      char* vps_base64 = base64Encode((char*)vps, vpsSize);
-      char* sps_base64 = base64Encode((char*)sps, spsSize);
-      char* pps_base64 = base64Encode((char*)pps, ppsSize);
+      char* vps_base64 = base64Encode((char *)(uintptr_t)vps, vpsSize);
+      char* sps_base64 = base64Encode((char *)(uintptr_t)sps, spsSize);
+      char* pps_base64 = base64Encode((char *)(uintptr_t)pps, ppsSize);
       delete[] vps; delete[] sps; delete[] pps;
 
       result = H265VideoFileSink::createNew(envir(), fileName,
@@ -812,13 +813,13 @@
 } 
 
 void MatroskaTrackTable::add(MatroskaTrack* newTrack, unsigned trackNumber) {
-  if (newTrack != NULL && newTrack->trackNumber != 0) fTable->Remove((char const*)newTrack->trackNumber);
-  MatroskaTrack* existingTrack = (MatroskaTrack*)fTable->Add((char const*)trackNumber, newTrack);
+  if (newTrack != NULL && newTrack->trackNumber != 0) fTable->Remove((char const*)(uintptr_t)newTrack->trackNumber);
+  MatroskaTrack* existingTrack = (MatroskaTrack*)fTable->Add((char const*)(uintptr_t)trackNumber, newTrack);
   delete existingTrack; // in case it wasn't NULL
 }
 
 MatroskaTrack* MatroskaTrackTable::lookup(unsigned trackNumber) {
-  return (MatroskaTrack*)fTable->Lookup((char const*)trackNumber);
+  return (MatroskaTrack*)fTable->Lookup((char const*)(uintptr_t)trackNumber);
 }
 
 unsigned MatroskaTrackTable::numTracks() const { return fTable->numEntries(); }
@@ -906,18 +907,18 @@
   if (trackNumber == 0) return NULL;
 
   FramedSource* trackSource = new MatroskaDemuxedTrack(envir(), trackNumber, *this);
-  fDemuxedTracksTable->Add((char const*)trackNumber, trackSource);
+  fDemuxedTracksTable->Add((char const*)(uintptr_t)trackNumber, trackSource);
   return trackSource;
 }
 
 MatroskaDemuxedTrack* MatroskaDemux::lookupDemuxedTrack(unsigned trackNumber) {
-  return (MatroskaDemuxedTrack*)fDemuxedTracksTable->Lookup((char const*)trackNumber);
+  return (MatroskaDemuxedTrack*)fDemuxedTracksTable->Lookup((char const*)(uintptr_t)trackNumber);
 }
 
 void MatroskaDemux::removeTrack(unsigned trackNumber) {
   if (fOurParser != NULL) fOurParser->stopAnyDeliveryForTrack(trackNumber);
 
-  fDemuxedTracksTable->Remove((char const*)trackNumber);
+  fDemuxedTracksTable->Remove((char const*)(uintptr_t)trackNumber);
   if (fDemuxedTracksTable->numEntries() == 0) {
     // We no longer have any demuxed tracks, so delete ourselves now:
     Medium::close(this);
diff -Naur live.orig/liveMedia/MediaSession.cpp live.step1/liveMedia/MediaSession.cpp
--- live.orig/liveMedia/MediaSession.cpp	2025-11-07 05:17:38.000000000 +0900
+++ live.step1/liveMedia/MediaSession.cpp	2025-11-30 15:19:48.724247165 +0900
@@ -20,6 +20,7 @@
 // Implementation
 
 #include "liveMedia.hh"
+#include <stdint.h>
 #include "Locale.hh"
 #include "Base64.hh"
 #include "GroupsockHelper.hh"
@@ -71,7 +72,7 @@
   const unsigned maxCNAMElen = 100;
   char CNAME[maxCNAMElen+1];
 #ifndef CRIS
-  gethostname((char*)CNAME, maxCNAMElen);
+  gethostname((char *)(uintptr_t)CNAME, maxCNAMElen);
 #else
   // "gethostname()" isn't defined for this platform
   sprintf(CNAME, "unknown host %d", (unsigned)(our_random()*0x7FFFFFFF));
@@ -164,13 +165,13 @@
       // This "m=" line is bad; output an error message saying so:
       char* sdpLineStr;
       if (nextSDPLine == NULL) {
-	sdpLineStr = (char*)sdpLine;
+	sdpLineStr = (char *)(uintptr_t)sdpLine;
       } else {
 	sdpLineStr = strDup(sdpLine);
 	sdpLineStr[nextSDPLine-sdpLine] = '\0';
       }
       envir() << "Bad SDP \"m=\" line: " <<  sdpLineStr << "\n";
-      if (sdpLineStr != (char*)sdpLine) delete[] sdpLineStr;
+      if (sdpLineStr != (char *)(uintptr_t)sdpLine) delete[] sdpLineStr;
 
       delete[] mediumName;
       delete subsession;
@@ -800,7 +801,7 @@
 	if ((fClientPortNum&1) != 0) { // it's odd
 	  // Record this socket in our table, and keep trying:
 	  unsigned key = (unsigned)fClientPortNum;
-	  Groupsock* existing = (Groupsock*)socketHashTable->Add((char const*)key, fRTPSocket);
+	  Groupsock* existing = (Groupsock*)socketHashTable->Add((char const*)(uintptr_t)key, fRTPSocket);
 	  delete existing; // in case it wasn't NULL
 	  continue;
 	}
@@ -822,7 +823,7 @@
 
 	  // Record the first socket in our table, and keep trying:
 	  unsigned key = (unsigned)fClientPortNum;
-	  Groupsock* existing = (Groupsock*)socketHashTable->Add((char const*)key, fRTPSocket);
+	  Groupsock* existing = (Groupsock*)socketHashTable->Add((char const*)(uintptr_t)key, fRTPSocket);
 	  delete existing; // in case it wasn't NULL
 	  continue;
 	}
diff -Naur live.orig/liveMedia/OggFile.cpp live.step1/liveMedia/OggFile.cpp
--- live.orig/liveMedia/OggFile.cpp	2025-11-07 05:17:38.000000000 +0900
+++ live.step1/liveMedia/OggFile.cpp	2025-11-30 15:19:48.724247165 +0900
@@ -19,6 +19,7 @@
 // Implementation
 
 #include "OggFileParser.hh"
+#include <stdint.h>
 #include "OggDemuxedTrack.hh"
 #include "ByteStreamFileSource.hh"
 #include "VorbisAudioRTPSink.hh"
@@ -71,15 +72,15 @@
   demuxRecord->onDeletionFunc = onDeletionFunc;
   demuxRecord->objectToNotify = objectToNotify;
 
-  fDemuxesTable->Add((char const*)demux, demuxRecord);
+  fDemuxesTable->Add((char const*)(uintptr_t)demux, demuxRecord);
 
   return demux;
 }
 
 void OggFile::removeDemux(OggDemux* demux) {
-  DemuxRecord* demuxRecord = (DemuxRecord*)(fDemuxesTable->Lookup((char const*)demux));
+  DemuxRecord* demuxRecord = (DemuxRecord*)(fDemuxesTable->Lookup((char const*)(uintptr_t)demux));
   if (demuxRecord != NULL) {
-    fDemuxesTable->Remove((char const*)demux);
+    fDemuxesTable->Remove((char const*)(uintptr_t)demux);
 
     if (demuxRecord->onDeletionFunc != NULL) {
       (*demuxRecord->onDeletionFunc)(demuxRecord->objectToNotify, demux);
@@ -176,7 +177,7 @@
   delete fDemuxesTable;
   delete fTrackTable;
 
-  delete[] (char*)fFileName;
+  delete[] (char *)(uintptr_t)fFileName;
 }
 
 void OggFile::handleEndOfBosPageParsing(void* clientData) {
@@ -213,12 +214,12 @@
 
 void OggTrackTable::add(OggTrack* newTrack) {
   OggTrack* existingTrack
-    = (OggTrack*)fTable->Add((char const*)newTrack->trackNumber, newTrack);
+    = (OggTrack*)fTable->Add((char const*)(uintptr_t)newTrack->trackNumber, newTrack);
   delete existingTrack; // if any
 }
 
 OggTrack* OggTrackTable::lookup(u_int32_t trackNumber) {
-  return (OggTrack*)fTable->Lookup((char const*)trackNumber);
+  return (OggTrack*)fTable->Lookup((char const*)(uintptr_t)trackNumber);
 }
 
 unsigned OggTrackTable::numTracks() const { return fTable->numEntries(); }
@@ -270,7 +271,7 @@
 
   resultTrackNumber = nextTrack->trackNumber;
   FramedSource* trackSource = new OggDemuxedTrack(envir(), resultTrackNumber, *this);
-  fDemuxedTracksTable->Add((char const*)resultTrackNumber, trackSource);
+  fDemuxedTracksTable->Add((char const*)(uintptr_t)resultTrackNumber, trackSource);
   return trackSource;
 }
 
@@ -278,12 +279,12 @@
   if (trackNumber == 0) return NULL;
 
   FramedSource* trackSource = new OggDemuxedTrack(envir(), trackNumber, *this);
-  fDemuxedTracksTable->Add((char const*)trackNumber, trackSource);
+  fDemuxedTracksTable->Add((char const*)(uintptr_t)trackNumber, trackSource);
   return trackSource;
 }
 
 OggDemuxedTrack* OggDemux::lookupDemuxedTrack(u_int32_t trackNumber) {
-  return (OggDemuxedTrack*)fDemuxedTracksTable->Lookup((char const*)trackNumber);
+  return (OggDemuxedTrack*)fDemuxedTracksTable->Lookup((char const*)(uintptr_t)trackNumber);
 }
 
 OggDemux::OggDemux(OggFile& ourFile)
@@ -309,7 +310,7 @@
 }
 
 void OggDemux::removeTrack(u_int32_t trackNumber) {
-  fDemuxedTracksTable->Remove((char const*)trackNumber);
+  fDemuxedTracksTable->Remove((char const*)(uintptr_t)trackNumber);
   if (fDemuxedTracksTable->numEntries() == 0) {
     // We no longer have any demuxed tracks, so delete ourselves now:
     Medium::close(this);
diff -Naur live.orig/liveMedia/OnDemandServerMediaSubsession.cpp live.step1/liveMedia/OnDemandServerMediaSubsession.cpp
--- live.orig/liveMedia/OnDemandServerMediaSubsession.cpp	2025-11-07 05:17:38.000000000 +0900
+++ live.step1/liveMedia/OnDemandServerMediaSubsession.cpp	2025-11-30 15:19:48.724247165 +0900
@@ -20,6 +20,7 @@
 // Implementation
 
 #include "OnDemandServerMediaSubsession.hh"
+#include <stdint.h>
 #include <GroupsockHelper.hh>
 
 OnDemandServerMediaSubsession
@@ -233,7 +234,7 @@
   } else { // TCP
     destinations = new Destinations(tcpSocketNum, rtpChannelId, rtcpChannelId, tlsState);
   }
-  fDestinationsHashTable->Add((char const*)clientSessionId, destinations);
+  fDestinationsHashTable->Add((char const*)(uintptr_t)clientSessionId, destinations);
 }
 
 void OnDemandServerMediaSubsession::startStream(unsigned clientSessionId,
@@ -246,7 +247,7 @@
 						void* serverRequestAlternativeByteHandlerClientData) {
   StreamState* streamState = (StreamState*)streamToken;
   Destinations* destinations
-    = (Destinations*)(fDestinationsHashTable->Lookup((char const*)clientSessionId));
+    = (Destinations*)(fDestinationsHashTable->Lookup((char const*)(uintptr_t)clientSessionId));
   if (streamState != NULL) {
     streamState->startPlaying(destinations, clientSessionId,
 			      rtcpRRHandler, rtcpRRHandlerClientData,
@@ -370,9 +371,9 @@
 
   // Look up (and remove) the destinations for this client session:
   Destinations* destinations
-    = (Destinations*)(fDestinationsHashTable->Lookup((char const*)clientSessionId));
+    = (Destinations*)(fDestinationsHashTable->Lookup((char const*)(uintptr_t)clientSessionId));
   if (destinations != NULL) {
-    fDestinationsHashTable->Remove((char const*)clientSessionId);
+    fDestinationsHashTable->Remove((char const*)(uintptr_t)clientSessionId);
 
     // Stop streaming to these destinations:
     if (streamState != NULL) streamState->endPlaying(destinations, clientSessionId);
@@ -502,7 +503,7 @@
 	  rangeLine, // a=range:... (if present)
 	  auxSDPLine, // optional extra SDP line
 	  trackId()); // a=control:<track-id>
-  delete[] (char*)rangeLine; delete[] keyMgmtLine; delete[] rtpmapLine;
+  delete[] (char *)(uintptr_t)rangeLine; delete[] keyMgmtLine; delete[] rtpmapLine;
 
   delete[] fSDPLines; fSDPLines = strDup(sdpLines);
   delete[] sdpLines;
diff -Naur live.orig/liveMedia/PassiveServerMediaSubsession.cpp live.step1/liveMedia/PassiveServerMediaSubsession.cpp
--- live.orig/liveMedia/PassiveServerMediaSubsession.cpp	2025-11-07 05:17:38.000000000 +0900
+++ live.step1/liveMedia/PassiveServerMediaSubsession.cpp	2025-11-30 15:19:48.725247206 +0900
@@ -20,6 +20,7 @@
 // Implementation
 
 #include "PassiveServerMediaSubsession.hh"
+#include <stdint.h>
 #include <GroupsockHelper.hh>
 
 ////////// PassiveServerMediaSubsession //////////
@@ -135,7 +136,7 @@
 	    rangeLine, // a=range:... (if present)
 	    auxSDPLine, // optional extra SDP line
 	    trackId()); // a=control:<track-id>
-    delete[] (char*)rangeLine; delete[] keyMgmtLine; delete[] rtpmapLine;
+    delete[] (char *)(uintptr_t)rangeLine; delete[] keyMgmtLine; delete[] rtpmapLine;
 
     fSDPLines = strDup(sdpLines);
     delete[] sdpLines;
@@ -182,7 +183,7 @@
 
   // Make a record of this client's source - for RTCP RR handling:
   RTCPSourceRecord* source = new RTCPSourceRecord(clientAddress, clientRTCPPort);
-  fClientRTCPSourceRecords->Add((char const*)clientSessionId, source);
+  fClientRTCPSourceRecords->Add((char const*)(uintptr_t)clientSessionId, source);
 }
 
 void PassiveServerMediaSubsession::startStream(unsigned clientSessionId,
@@ -209,7 +210,7 @@
     fRTCPInstance->sendReport();
 
     // Set up the handler for incoming RTCP "RR" packets from this client:
-    RTCPSourceRecord* source = (RTCPSourceRecord*)(fClientRTCPSourceRecords->Lookup((char const*)clientSessionId));
+    RTCPSourceRecord* source = (RTCPSourceRecord*)(fClientRTCPSourceRecords->Lookup((char const*)(uintptr_t)clientSessionId));
     if (source != NULL) {
       fRTCPInstance->setSpecificRRHandler(source->addr, source->port,
 					  rtcpRRHandler, rtcpRRHandlerClientData);
@@ -236,13 +237,13 @@
 
 void PassiveServerMediaSubsession::deleteStream(unsigned clientSessionId, void*& /*streamToken*/) {
   // Lookup and remove the 'RTCPSourceRecord' for this client.  Also turn off RTCP "RR" handling:
-  RTCPSourceRecord* source = (RTCPSourceRecord*)(fClientRTCPSourceRecords->Lookup((char const*)clientSessionId));
+  RTCPSourceRecord* source = (RTCPSourceRecord*)(fClientRTCPSourceRecords->Lookup((char const*)(uintptr_t)clientSessionId));
   if (source != NULL) {
     if (fRTCPInstance != NULL) {
       fRTCPInstance->unsetSpecificRRHandler(source->addr, source->port);
     }
 
-    fClientRTCPSourceRecords->Remove((char const*)clientSessionId);
+    fClientRTCPSourceRecords->Remove((char const*)(uintptr_t)clientSessionId);
     delete source;
   }
 }
diff -Naur live.orig/liveMedia/RTSPServer.cpp live.step1/liveMedia/RTSPServer.cpp
--- live.orig/liveMedia/RTSPServer.cpp	2025-11-07 05:17:38.000000000 +0900
+++ live.step1/liveMedia/RTSPServer.cpp	2025-11-30 15:19:48.725247206 +0900
@@ -19,6 +19,7 @@
 // Implementation
 
 #include "RTSPServer.hh"
+#include <stdint.h>
 #include "RTSPCommon.hh"
 #include "RTSPRegisterSender.hh"
 #include "Base64.hh"
@@ -257,17 +258,17 @@
 void RTSPServer
 ::noteTCPStreamingOnSocket(int socketNum, RTSPClientSession* clientSession, unsigned trackNum) {
   streamingOverTCPRecord* sotcpCur
-    = (streamingOverTCPRecord*)fTCPStreamingDatabase->Lookup((char const*)socketNum);
+    = (streamingOverTCPRecord*)fTCPStreamingDatabase->Lookup((char const*)(uintptr_t)socketNum);
   streamingOverTCPRecord* sotcpNew
     = new streamingOverTCPRecord(clientSession->fOurSessionId, trackNum, sotcpCur);
-  fTCPStreamingDatabase->Add((char const*)socketNum, sotcpNew);
+  fTCPStreamingDatabase->Add((char const*)(uintptr_t)socketNum, sotcpNew);
 }
 
 void RTSPServer
 ::unnoteTCPStreamingOnSocket(int socketNum, RTSPClientSession* clientSession, unsigned trackNum) {
   if (socketNum < 0) return;
   streamingOverTCPRecord* sotcpHead
-    = (streamingOverTCPRecord*)fTCPStreamingDatabase->Lookup((char const*)socketNum);
+    = (streamingOverTCPRecord*)fTCPStreamingDatabase->Lookup((char const*)(uintptr_t)socketNum);
   if (sotcpHead == NULL) return;
 
   // Look for a record of the (session,track); remove it if found:
@@ -288,10 +289,10 @@
 
     if (sotcpHead == NULL) {
       // There were no more entries on the list.  Remove the original entry from the hash table:
-      fTCPStreamingDatabase->Remove((char const*)socketNum);
+      fTCPStreamingDatabase->Remove((char const*)(uintptr_t)socketNum);
     } else {
       // Add the rest of the list into the hash table (replacing the original):
-      fTCPStreamingDatabase->Add((char const*)socketNum, sotcpHead);
+      fTCPStreamingDatabase->Add((char const*)(uintptr_t)socketNum, sotcpHead);
     }
   } else {
     // We found it on the list, but not at the head.  Unlink it:
@@ -304,7 +305,7 @@
 void RTSPServer::stopTCPStreamingOnSocket(int socketNum) {
   // Close any stream that is streaming over "socketNum" (using RTP/RTCP-over-TCP streaming):
   streamingOverTCPRecord* sotcp
-    = (streamingOverTCPRecord*)fTCPStreamingDatabase->Lookup((char const*)socketNum);
+    = (streamingOverTCPRecord*)fTCPStreamingDatabase->Lookup((char const*)(uintptr_t)socketNum);
   if (sotcp != NULL) {
     do {
       RTSPClientSession* clientSession
@@ -318,7 +319,7 @@
       delete sotcp;
       sotcp = sotcpNext;
     } while (sotcp != NULL);
-    fTCPStreamingDatabase->Remove((char const*)socketNum);
+    fTCPStreamingDatabase->Remove((char const*)(uintptr_t)socketNum);
   }
 
   RTPInterface::clearServerRequestAlternativeByteHandler(envir(), socketNum);
@@ -352,7 +353,7 @@
 // Handler routines for specific RTSP commands:
 
 void RTSPServer::RTSPClientConnection::handleCmd_OPTIONS() {
-  snprintf((char*)fResponseBuffer, sizeof fResponseBuffer,
+  snprintf((char *)(uintptr_t)fResponseBuffer, sizeof fResponseBuffer,
 	   "RTSP/1.0 200 OK\r\nCSeq: %s\r\n%sPublic: %s\r\n\r\n",
 	   fCurrentCSeq, dateHeader(), fOurRTSPServer.allowedCommandNames());
 }
@@ -427,7 +428,7 @@
     // (which is necessary to ensure that the correct URL gets used in subsequent "SETUP" requests).
     rtspURL = fOurRTSPServer.rtspURL(session, fClientInputSocket);
     
-    snprintf((char*)fResponseBuffer, sizeof fResponseBuffer,
+    snprintf((char *)(uintptr_t)fResponseBuffer, sizeof fResponseBuffer,
 	     "RTSP/1.0 200 OK\r\nCSeq: %s\r\n"
 	     "%s"
 	     "Content-Base: %s/\r\n"
@@ -477,20 +478,20 @@
 
 void RTSPServer::RTSPClientConnection::handleCmd_bad() {
   // Don't do anything with "fCurrentCSeq", because it might be nonsense
-  snprintf((char*)fResponseBuffer, sizeof fResponseBuffer,
+  snprintf((char *)(uintptr_t)fResponseBuffer, sizeof fResponseBuffer,
 	   "RTSP/1.0 400 Bad Request\r\n%sAllow: %s\r\n\r\n",
 	   dateHeader(), fOurRTSPServer.allowedCommandNames());
 }
 
 void RTSPServer::RTSPClientConnection::handleCmd_notSupported() {
-  snprintf((char*)fResponseBuffer, sizeof fResponseBuffer,
+  snprintf((char *)(uintptr_t)fResponseBuffer, sizeof fResponseBuffer,
 	   "RTSP/1.0 405 Method Not Allowed\r\nCSeq: %s\r\n%sAllow: %s\r\n\r\n",
 	   fCurrentCSeq, dateHeader(), fOurRTSPServer.allowedCommandNames());
 }
 
 void RTSPServer::RTSPClientConnection::handleCmd_redirect(char const* urlSuffix) {
   char* urlPrefix = fOurRTSPServer.rtspURLPrefix(fClientInputSocket);
-  snprintf((char*)fResponseBuffer, sizeof fResponseBuffer,
+  snprintf((char *)(uintptr_t)fResponseBuffer, sizeof fResponseBuffer,
 	   "RTSP/1.0 301 Moved Permanently\r\n"
 	   "CSeq: %s\r\n"
 	   "%s"
@@ -519,7 +520,7 @@
 								 char* acceptStr, unsigned acceptStrMaxSize) {
   // Check for the limited HTTP requests that we expect for specifying RTSP-over-HTTP tunneling.
   // This parser is currently rather dumb; it should be made smarter #####
-  char const* reqStr = (char const*)fRequestBuffer;
+  char const* reqStr = (char const*)(uintptr_t)fRequestBuffer;
   unsigned const reqStrSize = fRequestBytesAlreadySeen;
   
   // Read everything up to the first space as the command name:
@@ -567,13 +568,13 @@
 }
 
 void RTSPServer::RTSPClientConnection::handleHTTPCmd_notSupported() {
-  snprintf((char*)fResponseBuffer, sizeof fResponseBuffer,
+  snprintf((char *)(uintptr_t)fResponseBuffer, sizeof fResponseBuffer,
 	   "HTTP/1.0 405 Method Not Allowed\r\n%s\r\n\r\n",
 	   dateHeader());
 }
 
 void RTSPServer::RTSPClientConnection::handleHTTPCmd_notFound() {
-  snprintf((char*)fResponseBuffer, sizeof fResponseBuffer,
+  snprintf((char *)(uintptr_t)fResponseBuffer, sizeof fResponseBuffer,
 	   "HTTP/1.0 404 Not Found\r\n%s\r\n\r\n",
 	   dateHeader());
 }
@@ -583,7 +584,7 @@
   fprintf(stderr, "Handled HTTP \"OPTIONS\" request\n");
 #endif
   // Construct a response to the "OPTIONS" command that notes that our special headers (for RTSP-over-HTTP tunneling) are allowed:
-  snprintf((char*)fResponseBuffer, sizeof fResponseBuffer,
+  snprintf((char *)(uintptr_t)fResponseBuffer, sizeof fResponseBuffer,
 	   "HTTP/1.0 200 OK\r\n"
 	   "%s"
 	   "Access-Control-Allow-Origin: *\r\n"
@@ -601,13 +602,13 @@
     fOurRTSPServer.fClientConnectionsForHTTPTunneling = HashTable::create(STRING_HASH_KEYS);
   }
   delete[] fOurSessionCookie; fOurSessionCookie = strDup(sessionCookie);
-  fOurRTSPServer.fClientConnectionsForHTTPTunneling->Add(sessionCookie, (void*)this);
+  fOurRTSPServer.fClientConnectionsForHTTPTunneling->Add(sessionCookie, (void *)(uintptr_t)this);
 #ifdef DEBUG
   fprintf(stderr, "Handled HTTP \"GET\" request (client output socket: %d)\n", fClientOutputSocket);
 #endif
   
   // Construct our response:
-  snprintf((char*)fResponseBuffer, sizeof fResponseBuffer,
+  snprintf((char *)(uintptr_t)fResponseBuffer, sizeof fResponseBuffer,
 	   "HTTP/1.0 200 OK\r\n"
 	   "%s"
 	   "Cache-Control: no-cache\r\n"
@@ -790,7 +791,7 @@
     Boolean urlIsRTSPS;
     Boolean playAfterSetup = False;
     fLastCRLF[2] = '\0'; // temporarily, for parsing
-    Boolean parseSucceeded = parseRTSPRequestString((char*)fRequestBuffer, fLastCRLF+2 - fRequestBuffer,
+    Boolean parseSucceeded = parseRTSPRequestString((char *)(uintptr_t)fRequestBuffer, fLastCRLF+2 - fRequestBuffer,
 						    cmdName, sizeof cmdName,
 						    urlPreSuffix, sizeof urlPreSuffix,
 						    urlSuffix, sizeof urlSuffix,
@@ -848,14 +849,14 @@
       } else if (urlPreSuffix[0] == '\0' && urlSuffix[0] == '*' && urlSuffix[1] == '\0') {
 	// The special "*" URL means: an operation on the entire server.  This works only for GET_PARAMETER and SET_PARAMETER:
 	if (strcmp(cmdName, "GET_PARAMETER") == 0) {
-	  handleCmd_GET_PARAMETER((char const*)fRequestBuffer);
+	  handleCmd_GET_PARAMETER((char const*)(uintptr_t)fRequestBuffer);
 	} else if (strcmp(cmdName, "SET_PARAMETER") == 0) {
-	  handleCmd_SET_PARAMETER((char const*)fRequestBuffer);
+	  handleCmd_SET_PARAMETER((char const*)(uintptr_t)fRequestBuffer);
 	} else {
 	  handleCmd_notSupported();
 	}
       } else if (strcmp(cmdName, "DESCRIBE") == 0) {
-	handleCmd_DESCRIBE(urlPreSuffix, urlSuffix, (char const*)fRequestBuffer);
+	handleCmd_DESCRIBE(urlPreSuffix, urlSuffix, (char const*)(uintptr_t)fRequestBuffer);
       } else if (strcmp(cmdName, "SETUP") == 0) {
 	Boolean areAuthenticated = True;
 
@@ -872,7 +873,7 @@
 	    strcat(urlTotalSuffix, "/");
 	  }
 	  strcat(urlTotalSuffix, urlSuffix);
-	  if (authenticationOK("SETUP", urlTotalSuffix, (char const*)fRequestBuffer)) {
+	  if (authenticationOK("SETUP", urlTotalSuffix, (char const*)(uintptr_t)fRequestBuffer)) {
 	    clientSession
 	      = (RTSPServer::RTSPClientSession*)fOurRTSPServer.createNewClientSessionWithId();
 	  } else {
@@ -880,7 +881,7 @@
 	  }
 	}
 	if (clientSession != NULL) {
-	  clientSession->handleCmd_SETUP(this, urlPreSuffix, urlSuffix, (char const*)fRequestBuffer);
+	  clientSession->handleCmd_SETUP(this, urlPreSuffix, urlSuffix, (char const*)(uintptr_t)fRequestBuffer);
 	  playAfterSetup = clientSession->fStreamAfterSETUP;
 	} else if (areAuthenticated) {
 #ifdef DEBUG
@@ -894,7 +895,7 @@
 		 || strcmp(cmdName, "GET_PARAMETER") == 0
 		 || strcmp(cmdName, "SET_PARAMETER") == 0) {
 	if (clientSession != NULL) {
-	  clientSession->handleCmd_withinSession(this, cmdName, urlPreSuffix, urlSuffix, (char const*)fRequestBuffer);
+	  clientSession->handleCmd_withinSession(this, cmdName, urlPreSuffix, urlSuffix, (char const*)(uintptr_t)fRequestBuffer);
 	} else {
 #ifdef DEBUG
 	  fprintf(stderr, "Calling handleCmd_sessionNotFound() (case 3)\n");
@@ -904,14 +905,14 @@
       } else if (strcmp(cmdName, "REGISTER") == 0 || strcmp(cmdName, "DEREGISTER") == 0) {
 	// Because - unlike other commands - an implementation of this command needs
 	// the entire URL, we re-parse the command to get it:
-	char* url = strDupSize((char*)fRequestBuffer);
-	if (sscanf((char*)fRequestBuffer, "%*s %s", url) == 1) {
+	char* url = strDupSize((char *)(uintptr_t)fRequestBuffer);
+	if (sscanf((char *)(uintptr_t)fRequestBuffer, "%*s %s", url) == 1) {
 	  // Check for special command-specific parameters in a "Transport:" header:
 	  Boolean reuseConnection, deliverViaTCP;
 	  char* proxyURLSuffix;
-	  parseTransportHeaderForREGISTER((const char*)fRequestBuffer, reuseConnection, deliverViaTCP, proxyURLSuffix);
+	  parseTransportHeaderForREGISTER((const char *)(uintptr_t)fRequestBuffer, reuseConnection, deliverViaTCP, proxyURLSuffix);
 
-	  handleCmd_REGISTER(cmdName, url, urlSuffix, (char const*)fRequestBuffer, reuseConnection, deliverViaTCP, proxyURLSuffix);
+	  handleCmd_REGISTER(cmdName, url, urlSuffix, (char const*)(uintptr_t)fRequestBuffer, reuseConnection, deliverViaTCP, proxyURLSuffix);
 	  delete[] proxyURLSuffix;
 	} else {
 	  handleCmd_bad();
@@ -948,7 +949,7 @@
 	  if (strcmp(acceptStr, "application/x-rtsp-tunnelled") == 0) {
 	    isValidHTTPCmd = False;
 	  } else {
-	    handleHTTPCmd_StreamingGET(urlSuffix, (char const*)fRequestBuffer);
+	    handleHTTPCmd_StreamingGET(urlSuffix, (char const*)(uintptr_t)fRequestBuffer);
 	  }
 	} else if (strcmp(cmdName, "GET") == 0) {
 	  handleHTTPCmd_TunnelingGET(sessionCookie);
@@ -979,17 +980,17 @@
 #ifdef DEBUG
     fprintf(stderr, "sending response: %s", fResponseBuffer);
 #endif
-    unsigned const numBytesToWrite = strlen((char*)fResponseBuffer);
+    unsigned const numBytesToWrite = strlen((char *)(uintptr_t)fResponseBuffer);
     if (fOutputTLS->isNeeded) {
-        fOutputTLS->write((char const*)fResponseBuffer, numBytesToWrite);
+        fOutputTLS->write((char const*)(uintptr_t)fResponseBuffer, numBytesToWrite);
     } else {
-        send(fClientOutputSocket, (char const*)fResponseBuffer, numBytesToWrite, MSG_NOSIGNAL);
+        send(fClientOutputSocket, (char const*)(uintptr_t)fResponseBuffer, numBytesToWrite, MSG_NOSIGNAL);
    }
     
     if (playAfterSetup) {
       // The client has asked for streaming to commence now, rather than after a
       // subsequent "PLAY" command.  So, simulate the effect of a "PLAY" command:
-      clientSession->handleCmd_withinSession(this, "PLAY", urlPreSuffix, urlSuffix, (char const*)fRequestBuffer);
+      clientSession->handleCmd_withinSession(this, "PLAY", urlPreSuffix, urlSuffix, (char const*)(uintptr_t)fRequestBuffer);
     }
     
     // Check whether there are extra bytes remaining in the buffer, after the end of the request (a rare case).
@@ -1117,8 +1118,8 @@
     fCurrentAuthenticator.reclaimDigestResponse(ourResponse);
   } while (0);
   
-  delete[] (char*)realm; delete[] (char*)nonce;
-  delete[] (char*)uri; delete[] (char*)response;
+  delete[] (char *)(uintptr_t)realm; delete[] (char *)(uintptr_t)nonce;
+  delete[] (char *)(uintptr_t)uri; delete[] (char *)(uintptr_t)response;
   
   if (success) {
     // The user has been authenticated.
@@ -1127,18 +1128,18 @@
       // Note: We don't return a "WWW-Authenticate" header here, because the user is valid,
       // even though the server has decided that they should not have access.
       setRTSPResponse("401 Unauthorized");
-      delete[] (char*)username;
+      delete[] (char *)(uintptr_t)username;
       return False;
     }
   }
-  delete[] (char*)username;
+  delete[] (char *)(uintptr_t)username;
   if (success) return True;
   
   // If we get here, we failed to authenticate the user.
   // Send back a "401 Unauthorized" response, with a new random nonce:
   Boolean isInitial401 = fCurrentAuthenticator.nonce() == NULL;
   fCurrentAuthenticator.setRealmAndRandomNonce(authDB->realm());
-  snprintf((char*)fResponseBuffer, sizeof fResponseBuffer,
+  snprintf((char *)(uintptr_t)fResponseBuffer, sizeof fResponseBuffer,
 	   "RTSP/1.0 401 Unauthorized\r\n"
 	   "CSeq: %s\r\n"
 	   "%s"
@@ -1154,7 +1155,7 @@
 
 void RTSPServer::RTSPClientConnection
 ::setRTSPResponse(char const* responseStr) {
-  snprintf((char*)fResponseBuffer, sizeof fResponseBuffer,
+  snprintf((char *)(uintptr_t)fResponseBuffer, sizeof fResponseBuffer,
 	   "RTSP/1.0 %s\r\n"
 	   "CSeq: %s\r\n"
 	   "%s\r\n",
@@ -1165,7 +1166,7 @@
 
 void RTSPServer::RTSPClientConnection
 ::setRTSPResponse(char const* responseStr, u_int32_t sessionId) {
-  snprintf((char*)fResponseBuffer, sizeof fResponseBuffer,
+  snprintf((char *)(uintptr_t)fResponseBuffer, sizeof fResponseBuffer,
 	   "RTSP/1.0 %s\r\n"
 	   "CSeq: %s\r\n"
 	   "%s"
@@ -1181,7 +1182,7 @@
   if (contentStr == NULL) contentStr = "";
   unsigned const contentLen = strlen(contentStr);
   
-  snprintf((char*)fResponseBuffer, sizeof fResponseBuffer,
+  snprintf((char *)(uintptr_t)fResponseBuffer, sizeof fResponseBuffer,
 	   "RTSP/1.0 %s\r\n"
 	   "CSeq: %s\r\n"
 	   "%s"
@@ -1199,7 +1200,7 @@
   if (contentStr == NULL) contentStr = "";
   unsigned const contentLen = strlen(contentStr);
   
-  snprintf((char*)fResponseBuffer, sizeof fResponseBuffer,
+  snprintf((char *)(uintptr_t)fResponseBuffer, sizeof fResponseBuffer,
 	   "RTSP/1.0 %s\r\n"
 	   "CSeq: %s\r\n"
 	   "%s"
@@ -1592,7 +1593,7 @@
     if (fIsMulticast) {
       switch (streamingMode) {
           case RTP_UDP: {
-	    snprintf((char*)fOurClientConnection->fResponseBuffer, sizeof fOurClientConnection->fResponseBuffer,
+	    snprintf((char *)(uintptr_t)fOurClientConnection->fResponseBuffer, sizeof fOurClientConnection->fResponseBuffer,
 		     "RTSP/1.0 200 OK\r\n"
 		     "CSeq: %s\r\n"
 		     "%s"
@@ -1611,7 +1612,7 @@
 	    break;
 	  }
           case RAW_UDP: {
-	    snprintf((char*)fOurClientConnection->fResponseBuffer, sizeof fOurClientConnection->fResponseBuffer,
+	    snprintf((char *)(uintptr_t)fOurClientConnection->fResponseBuffer, sizeof fOurClientConnection->fResponseBuffer,
 		     "RTSP/1.0 200 OK\r\n"
 		     "CSeq: %s\r\n"
 		     "%s"
@@ -1627,7 +1628,7 @@
     } else {
       switch (streamingMode) {
           case RTP_UDP: {
-	    snprintf((char*)fOurClientConnection->fResponseBuffer, sizeof fOurClientConnection->fResponseBuffer,
+	    snprintf((char *)(uintptr_t)fOurClientConnection->fResponseBuffer, sizeof fOurClientConnection->fResponseBuffer,
 		     "RTSP/1.0 200 OK\r\n"
 		     "CSeq: %s\r\n"
 		     "%s"
@@ -1644,7 +1645,7 @@
 	    if (!fOurRTSPServer.fAllowStreamingRTPOverTCP) {
 	      fOurClientConnection->handleCmd_unsupportedTransport();
 	    } else {
-	      snprintf((char*)fOurClientConnection->fResponseBuffer, sizeof fOurClientConnection->fResponseBuffer,
+	      snprintf((char *)(uintptr_t)fOurClientConnection->fResponseBuffer, sizeof fOurClientConnection->fResponseBuffer,
 		       "RTSP/1.0 200 OK\r\n"
 		       "CSeq: %s\r\n"
 		       "%s"
@@ -1658,7 +1659,7 @@
 	    break;
 	  }
           case RAW_UDP: {
-	    snprintf((char*)fOurClientConnection->fResponseBuffer, sizeof fOurClientConnection->fResponseBuffer,
+	    snprintf((char *)(uintptr_t)fOurClientConnection->fResponseBuffer, sizeof fOurClientConnection->fResponseBuffer,
 		     "RTSP/1.0 200 OK\r\n"
 		     "CSeq: %s\r\n"
 		     "%s"
@@ -1966,7 +1967,7 @@
   }
   
   // Fill in the response:
-  snprintf((char*)ourClientConnection->fResponseBuffer, sizeof ourClientConnection->fResponseBuffer,
+  snprintf((char *)(uintptr_t)ourClientConnection->fResponseBuffer, sizeof ourClientConnection->fResponseBuffer,
 	   "RTSP/1.0 200 OK\r\n"
 	   "CSeq: %s\r\n"
 	   "%s"
