diff -Naur live.step2/groupsock/NetAddress.cpp live.step3/groupsock/NetAddress.cpp
--- live.step2/groupsock/NetAddress.cpp	2025-11-07 05:17:38.000000000 +0900
+++ live.step3/groupsock/NetAddress.cpp	2025-11-30 15:20:51.572777412 +0900
@@ -157,7 +157,7 @@
       return ((struct sockaddr_in const&)left).sin_addr.s_addr == ((struct sockaddr_in const&)right).sin_addr.s_addr;
     }
     case AF_INET6: {
-      return ((struct sockaddr_in6 const&)left).sin6_addr.s6_addr == ((struct sockaddr_in6 const&)right).sin6_addr.s6_addr;
+      return +((struct sockaddr_in6 const&)left).sin6_addr.s6_addr == +((struct sockaddr_in6 const&)right).sin6_addr.s6_addr;
     }
     default: {
       return False;
